pkg/codegen/internal/test/testdata/simple-enum-schema/python/pulumi_plant_provider/_inputs.py

```diff
@@ -16,31 +16,31 @@
 @pulumi.input_type
 class ContainerArgs:
     def __init__(__self__, *,
-                 size: pulumi.Input[enums.ContainerSize],
-                 color: Optional[pulumi.Input[Union[enums.ContainerColor, str]]] = None,
+                 size: pulumi.Input[_enums._ContainerSize],
+                 color: Optional[pulumi.Input[Union[_enums._ContainerColor, str]]] = None,
                  material: Optional[pulumi.Input[str]] = None):
-        pulumi.set(__self__, "size", size)
+        pulumi.set(__self__, "size", size.value if isinstance(size, Enum) else size)
```

Comment:
Going to move this typecheck into `pulumi.set()` itself.

Reply:
I suspect this isn't going to work in all cases. This is an `Input[T]`, so it _may_ be `T` (where this would work), but it could also be `Awaitable[T]` or `Output[T]`. I think we're going to have to just pass the value as-is and add this logic inside [`rpc.serialize_property`](https://github.com/pulumi/pulumi/blob/eaac9280f5484f94213dbf54c9fd6f009d87db12/sdk/python/lib/pulumi/runtime/rpc.py#L88).